### PR TITLE
Fix net 8 native workflows

### PR DESCRIPTION
**Pull request by Beau-Gosse-dev**
*Description of changes:*
Mostly fixing places where `native` was missed. Once this deploys, then we can get the API URL to set the secret. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/net-8-native-aot-pipeline-canary.yaml [hunk 1]
```diff
@@ -13,26 +13,26 @@ jobs:
...
         - name: Build and Deploy
-          uses: ./.github/actions/sam-build-and-deploy
+          uses: ./.github/actions/sam-build-and-deploy-native
           with:
             aws-access-key: ${{ secrets.AWS_ACCESS_KEY_ID }}
             aws-secret-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
             aws-region: ${{ secrets.AWS_REGION }}
             dotnet-version: '8.x'
-            template-file-path: ./src/NET8/template.yaml
-            project-directory: ./src/NET8/
```
Comment (ppittle) [inline .github/workflows/net-8-native-aot-pipeline-canary.yaml:23]: Quick check: is `./src/NET8` still used?  Or is it fully replaced by `./src/NET8Native/`?

Comment (ppittle) [reply .github/workflows/net-8-native-aot-pipeline-canary.yaml:23]: If `./src/NET8` is no longer needed, can it be removed?

Comment (Beau-Gosse-dev) [reply .github/workflows/net-8-native-aot-pipeline-canary.yaml:23]: NET8 is still used for the non-native .NET 8 canary

Comment (Beau-Gosse-dev) [reply .github/workflows/net-8-native-aot-pipeline-canary.yaml:23]: This was just a copy/paste miss when creating the native canary from the non-native canary

Comment (Beau-Gosse-dev) [reply .github/workflows/net-8-native-aot-pipeline-canary.yaml:23]: It's used here: https://github.com/aws-samples/serverless-dotnet-demo/blob/main/.github/workflows/net-8-pipeline-canary.yaml

Comment (ppittle) [reply .github/workflows/net-8-native-aot-pipeline-canary.yaml:23]: Sounds good - just wanted to confirm.

**Review by ppittle — APPROVED** [review]

**Pull request merged**